python/mxnet/gluon/block.py

```diff
@@ -1343,13 +1343,18 @@ def export(self, path, epoch=0, remove_amp_cast=True):
         arg_names = set(sym.list_arguments())
         aux_names = set(sym.list_auxiliary_states())
         arg_dict = {}
-        for is_arg, name, param in self._cached_op_args:
-            if not is_arg:
-                if name in arg_names:
-                    arg_dict['arg:{}'.format(name)] = param._reduce()
-                else:
+        for name, param in self.collect_params().items():
```

Comment:
self.collect_params() does not contain all parameters if the block has been optimized for a backend. You need to rely on `self._cached_op_args`